Gitlab-146: Added test toggle for the setup script

shayfaber opened the pull request:
Added a test toggle that will run only the setup script if set to true, and skips it when set to false. This means you can now run `npx playwright test` without needing to exclude setup.spec.ts

shayfaber committed:
Added test toggle for the setup script

Added a test toggle that will run only the setup script if set to true,
and skips it when set to false. This means you can now run
`npx playwright test` without needing to exclude setup.spec.ts

dheesen commented on tests/base/config/test-toggles.json:
@shayfaber This file shouldn't be pushed to the repo. This is for local use only.

shayfaber replied:
Agreed, it's also set in .gitignore so I'm not sure why it got pushed.

dheesen commented on tests/base/setup.spec.ts:
```diff
@@ -1,110 +1,122 @@
-
-  const browserEngine = browserName?.toUpperCase() || "UNKNOWN";
-
+if(toggles.general.setup === true) {
```
@shayfaber You can shorten this as `(toggles.general.step)`

dheesen commented on tests/base/setup.spec.ts:
```diff
@@ -1,110 +1,122 @@
+        await magentoAdminPage.enableMultipleAdminLogins();
       } else {
-        throw new Error('.env file not found. Please ensure it exists in the root directory.');
+        testInfo.skip(true, `Skipping because configuration is only needed once.`);
```
@shayfaber The above throw of error was looking for .env, but now it's given a completely different error message. Is this right?

shayfaber replied:
I think this is correct! In lines 24-29, we confirm that the magento variables have been set in the `.env` files (I think this is the error that used to be here). Then, we check if the browser is Chromium. If it isn't we can just skip - there's no need to throw an error in that situation.

shayfaber committed:
Updated setup.spec.ts to always run in CI

shayfaber committed:
Delete tests/base/config/test-toggles.json

This file is added to .gitignore

shayfaber committed:
Update main.yml to copy test-toggles.example.json

Copies `/tests/base/config/test-toggles.example.json`

shayfaber committed:
Update main.yml (fixed mistake on line 53)

Line 53 (`cp tests/baseconfig/test-toggles.example.json [...]`) contained a mistake meaning Github Actions couldn't find the file to copy.

shayfaber committed:
Added check to `setup.spec.ts` to fix CI/other split

This rewrites `setup.spec.ts` to ensure the following:

1. `setup.spec.ts` always runs in CI
2. If not, we check if the setup test toggle is set to true.
3. If that's true, we run the setup.
